lib/filewatch/watch.rb

```diff
@@ -1,59 +1,36 @@
 require "logger"
+require_relative 'watched_file'
+
 if RbConfig::CONFIG['host_os'] =~ /mswin|mingw|cygwin/
   require "filewatch/winhelper"
+  INODE_METHOD = :win_inode
+else
+  INODE_METHOD = :nix_inode
 end
 
+MAX_OPEN_FILES = ENV.fetch("MAX_OPEN_FILES", 4095).to_i
```

Comment:
Its both. ENV for when the config is not available to the user.
There are precedents in using both: ``` ENV["SINCEDB_PATH"] ```

In watch.rb:
```ruby
...
    @max_active = MAX_OPEN_FILES
...
    def max_open_files=(value)
      return if value.nil?
      val = value.to_i
      @max_active = val <= 0 ? MAX_OPEN_FILES : value
    end
```
and in tail_base.rb:59
```ruby
    @watch.max_open_files = @opts[:max_open_files]
```